src/routes/article-list.ts

```diff
@@ -1,11 +1,12 @@
-import Router from '@koa/router';
-import { Context, Middleware, Next } from 'koa';
+import { Next } from 'koa';
 import { hydra, rdf, schema } from 'rdf-namespaces';
-import Articles from '../articles';
+import { AppContext, AppMiddleware } from '../types';
 import Routes from './index';
 
-export default (articles: Articles, router: Router): Middleware => (
-  async ({ request, response }: Context, next: Next): Promise<void> => {
+export default (): AppMiddleware => (
+  async ({
+    articles, request, response, router,
```

Comment:
it is a refactoring from Dependency Injection to a Service Locator (the `AppContext`), but at least this unpacking makes the dependencies clear